src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -86,14 +94,69 @@ private BoundExpression RewriteCollectionExpressionConversion(Conversion convers
                 _factory.Syntax = previousSyntax;
             }
 
-            // If the collection type is List<T> and items are added using the expected List<T>.Add(T) method,
-            // then construction can be optimized to use CollectionsMarshal methods.
-            static bool useListOptimization(CSharpCompilation compilation, BoundCollectionExpression node, out TypeWithAnnotations elementType)
+            // If we have something like `List<int> l = [.. someEnumerable]` try rewrite it using
+            // `Enumerable.ToList` or `new List(IEnumerable)` members if possible
+            static bool tryRewriteSingleElementSpreadToList(
```

Comment:
>tryRewriteSingleElementSpreadToList [](http://example.com/codeflow?start=24&length=35)

Let's make this an instance method rather than a local function. Then the parameters `compilation`, `self`, and `factory` would not be necessary.

Reply:
or just a non-static local function.

Reply:
Yes, it could be a non-static local function. But using an instance method would avoid unexpected capture.